### PR TITLE
initial dev user model

### PR DESCRIPTION
**Pull request by ericshuanders**
I've left out the 'next' parameter and 'next( )' function in the function that hashes the dev user password before saving.  I think this is ok since we are using async/await, but the todo app included the 'next' parameter and 'next()'.  I think it was redundant, so I nixed it, but feel free to double check me!

### CODE REVIEW

**Review by hugohp19 — COMMENTED** [review]
Hey, it looks like the JWT secret is in the sample.env file and it should be on the .env

**Comment by ericshuanders** [comment]
Good catch!  Moved it to .env now.  Also, added some more fields from our quiz.

File: server/db/models/Developer.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      trim: true,
+      unique: true,
+    },
+    github: {
+      type: String,
+      trim: true,
+      unique: true,
+    },
+    teamLead: {
+      type: String,
```
Comment (itchonib) [inline server/db/models/Developer.js]: Hello! looks like this might need to be a boolean?

File: server/db/models/Developer.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    email: {
+      type: String,
+      required: true,
+      trim: true,
+      unique: true,
+    },
+    school: {
+      type: String,
+      trim: true,
+      unique: true,
```
Comment (hugohp19) [inline server/db/models/Developer.js]: should the school be unique? if it's unique, then we wouldn't be able to have the same school on multiple developers, right?
same with teamlead (not sure what is teamLead, I am guessing it is related to role?)

Comment (ericshuanders) [reply server/db/models/Developer.js]: I think role was front-end/ back-end/ full-stack, where as teamLead would be whether or not a developer wanted to be a PM or not.  But yes, unique should not be a a truthy attribute for either of them!

**Review by itchonib — APPROVED** [review]

**Review by hugohp19 — APPROVED** [review]

**Comment by itchonib** [comment]
closes #1

**Pull request merged**